atlasdb-cassandra-multinode-tests/src/test/java/com/palantir/cassandra/multinode/NodesDownTestSetup.java

```diff
@@ -0,0 +1,158 @@
+            .with(new ThreeNodeCassandraCluster());
+
+    public static void initializeKvsAndDegradeCluster(List<String> nodesToKill) {
+        setupTestTable();
+        db = createCassandraKvs();
+        degradeCassandraCluster(nodesToKill);
+    }
+
+    @AfterClass
+    public static void closeKvs() throws IOException, InterruptedException {
+        db.close();
+    }
+
+    private static void setupTestTable() {
+        CassandraKeyValueService setupDb = createCassandraKvs();
+        setupDb.createTable(TEST_TABLE, AtlasDbConstants.GENERIC_TABLE_METADATA);
+        setupDb.put(TEST_TABLE, ImmutableMap.of(CELL_1_1, PtBytes.toBytes("old_value")), OLD_TIMESTAMP);
+        setupDb.put(TEST_TABLE, ImmutableMap.of(CELL_1_1, DEFAULT_CONTENTS), DEFAULT_TIMESTAMP);
+        setupDb.put(TEST_TABLE, ImmutableMap.of(CELL_1_2, DEFAULT_CONTENTS), DEFAULT_TIMESTAMP);
+        setupDb.put(TEST_TABLE, ImmutableMap.of(CELL_2_1, DEFAULT_CONTENTS), DEFAULT_TIMESTAMP);
+
+        setupDb.createTable(TEST_TABLE_TO_DROP, AtlasDbConstants.EMPTY_TABLE_METADATA);
+        setupDb.createTable(TEST_TABLE_TO_DROP_2, AtlasDbConstants.EMPTY_TABLE_METADATA);
+
+        setupDb.close();
+    }
+
+    protected static CassandraKeyValueService createCassandraKvs() {
+        CassandraKeyValueServiceConfig config = ImmutableCassandraKeyValueServiceConfig
+                .copyOf(ThreeNodeCassandraCluster.KVS_CONFIG)
+                .withSchemaMutationTimeoutMillis(3_000);
+        return CassandraKeyValueService.create(
+                CassandraKeyValueServiceConfigManager.createSimpleManager(config),
+                ThreeNodeCassandraCluster.LEADER_CONFIG);
+    }
+
+    private static void degradeCassandraCluster(List<String> nodesTokill) {
+        nodesTokill.forEach((containerName) -> {
+            try {
+                killCassandraContainer(containerName);
+            } catch (IOException | InterruptedException e) {
+                Throwables.propagate(e);
+            }
+        });
+
+        // startup checks aren't guaranteed to pass immediately after killing the node, so we wait until
+        // they do. unclear if this is an AtlasDB product problem. see #1154
+        if (nodesTokill.size() < 2) {
+            waitUntilStartupChecksPass();
+        }
+    }
+
+
+    private static void killCassandraContainer(String containerName) throws IOException, InterruptedException {
+        CONTAINERS.getContainer(containerName).kill();
+    }
+
+
+    private static void waitUntilStartupChecksPass() {
+        Awaitility.await()
+                .atMost(60, TimeUnit.SECONDS)
+                .until(NodesDownTestSetup::startupChecksPass);
+    }
+
+    private static boolean startupChecksPass() {
+        CassandraKeyValueServiceConfigManager manager = CassandraKeyValueServiceConfigManager.createSimpleManager(
+                ThreeNodeCassandraCluster.KVS_CONFIG);
+        try {
+            // startup checks are done implicitly in the constructor
```

Comment:
This is again a part of the refactor. Mostly making Test Setup common for one, two, three node down tests, we had only OneNodeDownTestSuite before. Not sure if the value will be seen in a separate PR.